platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/roster/RosterUtils.java

```diff
@@ -338,4 +338,29 @@ public static AddressBook buildAddressBook(@NonNull final Roster roster) {
 
         return addressBook;
     }
+
+    /**
+     * Verifies that all nodes are the same between the two specified rosters.
+     *
+     * @param roster1 the first roster to compare
+     * @param roster2 the second roster to compare
+     * @throws IllegalStateException if the rosters do not match
+     */
+    public static void verifyReconnectRosters(@NonNull final Roster roster1, @NonNull final Roster roster2)
```

Comment:
Minor: in theory, this method is currently fully equivalent to `roster1.equals(roster2)`. And I don't believe this is going to change in the future. Any other thoughts, @edward-swirldslabs ?

Perhaps we could nuke it and simply call the `equals` in reconnect code?